pkg/cmd/repo/create/http.go

```diff
@@ -104,34 +127,55 @@ func repoCreate(client *http.Client, hostname string, input repoCreateInput, tem
 		return api.InitRepoHostname(&response.CloneTemplateRepository.Repository, hostname), nil
 	}
 
-	var response struct {
-		CreateRepository struct {
-			Repository api.Repository
+	if input.GitIgnoreTemplate != "" || input.LicenseTemplate != "" {
+		inputv3 := createRepositoryInputV3{
+			Name:              input.Name,
+			HomepageURL:       input.HomepageURL,
+			Description:       input.Description,
+			IsPrivate:         strings.EqualFold(input.Visibility, "PRIVATE"),
+			TeamID:            teamIDv3,
+			HasIssuesEnabled:  input.HasIssuesEnabled,
+			HasWikiEnabled:    input.HasWikiEnabled,
+			GitIgnoreTemplate: input.GitIgnoreTemplate,
+			LicenseTemplate:   input.LicenseTemplate,
 		}
-	}
 
-	variables := map[string]interface{}{
-		"input": input,
-	}
+		path := "user/repos"
+		if isOrg {
+			path = fmt.Sprintf("orgs/%s/repos", input.OwnerLogin)
+			inputv3.Visibility = strings.ToLower(input.Visibility)
```

Comment:
This is not a CLI issue but the existence of `private` and `visibility` in this API endpoint is confusing.